### DISCUSSION TITLE
When is Livewire V3 to be released?

### DISCUSSION THREAD

**Discussion by flexchar** [livewire/livewire#5261] [1. Help]
I've seen the talk. It's amazing. But I cannot find it anywhere. When is Livewire V3 to be released?

The talk: https://www.youtube.com/watch?v=f4QShF42c6E&t=11907s

**Comment by ahosker** [comment]
Dates mean pressure, expectations and perhaps shortcuts. Though i'd be nice to have a Hot or Cold meter, to answer the question "start a project or wait"?.

**Reply by Wulfheart** [reply]
This!

**Reply by judgej** [reply]
Knowing dependencies means easier planning and fewer questions. For example, is the plan to wait for Laravel ~~9~~ *10!* to support that version only for new functionality? Or will it support version 9, perhaps with less functionality, or perhaps some Laravel 10 details need to be explored before knowing how Livewire is going to implement what is planned.

It would just be nice to know, because this whole framework ecosystem is conmplex, and projects need much planning :-)

Update: I'm not complaining or demanding, or expecting anything, and am most appreciative of everything we have got up to now. I'm just expressing what it feels like to be totally out of the loop.

**Reply by RealMrHex** [reply]
It is exactly like that; We have been waiting for a new update of Livewire for a long time and there is no definite schedule for it.
We were hoping to have a new version of Livewire at the beginning of the new year, but it didn't happen.
Perfectionism in updating the third version has caused a huge delay in between and this is in the case that if this big and perfectionistic process was transformed into smaller operations and presented in the form of specific updates, the appearance of the third version It didn't take that much time.

**Comment by joshhanley** [comment]
@flexchar @ahosker @Wulfheart currently there is no release date for V3. While Caleb demo'ed a lot of the new functionality working, there is still a fair bit to do to ensure it is feature complete from V2 features.

Due to that, it's hard to say even if it's hot/cold.

But a public beta will be released once it is getting close, and hopefully at that stage there will be a better idea of when it may be ready.

Personally I wouldn't be holding off on any new projects, I would start them on V2, then when V3 is released it should be a pretty straight forward upgrade.

Hope this helps!

**Reply by ariaieboy** [reply]
we need livewire 3 🥺

hope it get ready for beta soon

**Reply by joshhanley** [reply]
Please be patient, good things take time 🙂

**Reply by ariaieboy** [reply]
> Please be patient, good things take time 🙂

It's @calebporzio fault. 
After his talk on Laracon online, we are all excited about the new version of livewire 😊 and we can't wait for the public release.

**Reply by joshhanley** [reply]
@ariaieboy heh, yeah I know, I'm keen too 😁 I just know how big Livewire's codebase is and what's involved in re-writing it all 😅

**Comment by jeremiahsherrill** [comment]
Need some livewire 3... getting antsy over here. :)

**Comment by QasemiX** [comment]
Come on caleb, we're waiting........................................................................

**Comment by khalilsarabey** [comment]
I was hoping for v3 to be our Christmas gift! 
Now I'm hoping for it to be our new year's 😄

**Reply by thenibirahmed** [reply]
well, new year's past too...
let's see if it's valentines

**Reply by code23-isaac** [reply]
or april fools

**Comment by ariaieboy** [comment]
laravel 10 releases at February 7th, 2023 maybe livewire 3 release at the same time or after laravel 10 release ?
minimum laravel version gonna be 10 or 9 ?

**Reply by JohnnyMaynne** [reply]
<img width="703" alt="image" src="https://user-images.githubusercontent.com/26120256/214132619-b4b7ff61-50cc-4f3c-af6b-38bb185d46e5.png">

**Comment by Mohammed-Thamnees** [comment]
waiting........

**Comment by richardkeep** [comment]
waiting...

**Comment by krlenix** [comment]
Waiting...

**Comment by alekseyshavrak** [comment]
Waiting...

**Comment by CallMeNegar** [comment]
hell, still waiting and there is no news...

**Comment by webparadoxer** [comment]
no updates?

**Reply by richardkeep** [reply]
No Updates

**Comment by calebporzio** [comment]
Hey friends,

Still no date announcement, but I can tell you we're making lots of progress.

We have nearly all the unit and browser tests passing for v2 compatibility, so what's left now is: finishing up the v3 features (polishing them) and refactoring parts of the codebase.

We have an Alpha out right now to an internal group and as we make progress, we'll be launching a beta and letting people into it.

I know everyone wants V3 done yesterday, but I've decided to take the time I need to make sure it's as good as it can possibly be before getting locked into an official release.

Thanks for your patience!

**Reply by woodspire** [reply]
Would be nice if v3 could support multiple upload to Amazon S3.

**Reply by vlados** [reply]
I think you're team is making great things but this idea to not let the community help is strange. 
We are all a community and this makes Livewire and Laravel great. 

It will be good if we have a timetable or idea what is going to happen.

**Comment by code23-isaac** [comment]
Thanks for the update Caleb – any word on the future of alpine/livewire in regard to CSP?


Isaac Fennell
Web Developer
***@***.***
01189 306717

**Comment by jeremiahsherrill** [comment]
@calebporzio  “If you look for perfection, you'll never be content.”
― Leo Tolstoy

**Reply by joshhanley** [reply]
🙂 "Patience is a form of wisdom. It demonstrates that we understand and accept the fact that sometimes things must unfold in their own time." - Jon Kabat-Zinn

**Reply by tdondich** [reply]
"For me, I really like corn." - Corn kid

**Comment by CallMeNegar** [comment]


**Reply by Wulfheart** [reply]
Ehm... Well. Don't know how to say this. Did you pay for it? If not, what do they owe us?

**Reply by ariaieboy** [reply]
This kind of language is not allowed in the OSS community.

Yes, we all want to access the new version asap but that does not mean that we can talk like that.

As I know there is a close alpha that you can join by using the partnership program so if you want to get early access to the new versions you can join the partner program of a livewire and start supporting the project and get access faster than others.
https://www.laravel-livewire.com/support

**Comment by RealMrHex** [comment]
no updates? :(

**Reply by richardkeep** [reply]
We do https://github.com/livewire/livewire/discussions/5585

**Comment by manusiakemos** [comment]
can't wait 🥰

**Reply by forgecodex** [reply]
May be 2025 year

**Comment by mai1237** [comment]
Is there any chance to be known when arrives the release Livewire V3. At least any estimation of date?

**Comment by Wotan6891** [comment]
Gentlemen, you have to be patient. Developers who are responsible for innovating and updating open source software are not charging for giving the software to the community for free.

On the other hand, it would be good if they allowed the support of the community, since some from what I read want to help.

**Comment by joshhanley** [comment]
Have a look at this other discussion for more details [When V3?](https://github.com/livewire/livewire/discussions/5585)

**Comment by wbetalleluzd** [comment]
No news :(

**Comment by asterism612** [comment]
wating

**Comment by sort72** [comment]
:(

**Comment by jeremiahsherrill** [comment]
Somewhere beyond the commit stream,
Where bugs are scarce, and features gleam,
There's a version we dream of, oh, so grand,
Called Livewire Three, is it close at hand?

We're feeling low, our spirits are down,
As we scan the forums with a frown,
Waiting for Livewire's upgrade, oh so grand,
It's a melancholy we can barely stand.

Somewhere long past version two,
In the place where breakthroughs ensue,
Livewire Three, we yearn for thee,
To elevate our code in harmony.

Someday we'll wake and see the day,
When Livewire Three comes out to play,
With powerful bindings, real-time and more,
It's the upgrade that we're waiting for.

**Comment by tdondich** [comment]
What is the key feature in Laravel 3 that you are looking for?  Comment
here and if you see it plus it.  Let’s see what’s most in demand

**Reply by ariaieboy** [reply]
The latest Laravel version is 10, not 3 😄 

But about Livewire v3, I love all features that Caleb showed in his talk.

My personal choice is:
Alpine based core
Reactive props and $parent
livewire navigation
teleport
lazy components

**Reply by ariaieboy** [reply]
I have been using Livewire for a while now, and these new changes will fix 90% of my problems and headaches with Livewire.

**Comment by iammuttaqi** [comment]
Livewire V3 feels like the elusive unicorn we've been chasing for 9 months! Any chance it's secretly hiding in a magical realm? We're eagerly waiting for its arrival..

**Comment by muath-ye** [comment]
waiting too...

**Comment by allantinashemuzeya** [comment]
Wow its been 6 months and we are still waiting.....

**Comment by jeremiahsherrill** [comment]
Caleb really shouldn't have showcased livewire at last years laracon.

**Reply by ariaieboy** [reply]
+

**Reply by killjin** [reply]
Yes，8 months until now, no release time.

**Reply by RealMrHex** [reply]
+

**Comment by jeremiahsherrill** [comment]
Due to announcement I updated my poem

Somewhere beyond the commit stream,
Where bugs are scarce, and features gleam,
There's a version awaited, oh, so grand,
Livewire V3, soon it will land.

We're feeling low, our spirits may fall,
As we scan the forums, posts wall-to-wall,
Awaiting Livewire's upgrade, oh so grand,
Knowing its arrival is close at hand.

Somewhere beyond the chatter and clatter,
On July 20th, amidst Laracon's patter,
Livewire V3, our hearts beat for thee,
You'll elevate our code in harmony.

Soon we'll wake to the glorious day,
When Livewire V3 comes out to play,
With powerful bindings, real-time galore,
It's the upgrade we've been yearning for.

Post-Laracon talk, it will be free,
For us to explore, as happy as can be,
Livewire V3, we'll dance with glee,
Embracing the possibilities of a code spree.

**Comment by da-mask** [comment]
[https://v3when.com](https://v3when.com/)

**Comment by muath-ye** [comment]
Visit [v3when.com](https://v3when.com) to know when it will be released.

**Comment by sbc640964** [comment]
tomorrow!!!

**Comment by iammuttaqi** [comment]
today!

**Comment by marispro** [comment]
Beer prepared

**Comment by higorch** [comment]
![image](https://github.com/livewire/livewire/assets/13771352/ef8815f0-f328-4b7a-b89f-b0aa30835562)

**Comment by jeremiahsherrill** [comment]
It's today... where my livewire... where huh huh give it to me

**Comment by cmcneese** [comment]
> I've seen the talk. It's amazing. But I cannot find it anywhere. When is Livewire V3 to be released?
> 
> The talk: https://www.youtube.com/watch?v=f4QShF42c6E&t=11907s

It was JUST released today at LaraconUS in Nashville, TN. I just watched him launch it from the stage during his talk.

**Reply by jeremiahsherrill** [reply]
I am hoping they are planning to put it up on youtube at some point, I couldn't make it this year

**Reply by vlados** [reply]
https://www.youtube.com/watch?v=I0jhqcgOuvQ

**Reply by jeremiahsherrill** [reply]
ty damn it... I didn't see this advertised anywhere

**Comment by richardkeep** [comment]
https://livewire.laravel.com/

This issue has been closed.
